src/Microsoft.AspNet.Authentication/AuthenticationHandler.cs

```diff
@@ -155,14 +157,24 @@ public void GetDescriptions(DescribeSchemesContext describeContext)
             }
         }
 
+        protected Task<AuthenticationTicket> AuthenticateOnceAsync()
```

Comment:
You aren't supposed to return null tasks, the idea is only to call authenticate once, and subsequent calls should return the same result/ticket/null